File: Arrays/TwoSum/Go/1-Two-Sum.go
func twoSum(nums []int, target int) []int {

    nums_dict := make(map[int]int)

    for i, num := range nums{
        comp := target - nums[i]

        if j, exists := nums_dict[comp]; exists{
            return []int{j, i}
        }

        nums_dict[num] = i
    }

    return nil
    
}

File: Arrays/1-Two-Sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        nums_dict = {}

        for i, n in enumerate(nums):
            comp = target - n

            if comp in nums_dict:
                return [nums_dict[comp], i]
            
            nums_dict[n] = i
        return None
